Assets/Scripts/Battlefield/CreatureScripts/UniqueCreature.cs

```diff
@@ -1,81 +1,75 @@
 ﻿using System.Collections;
 using System.Collections.Generic;
 using UnityEngine;
+using Cinemachine;
 
-public class UniqueCreature : CreatureBase
-{
-    [Header("Ability Info")]
-    public Ability[] abilities;
-    public int AC;
-    public int hitMod;
-    public Material[] mat;
-    Renderer currentMat;
-    float a = .05f;
-    float b;
-    int highlightColor;
-
-    void Start()
+namespace SwordAndBored.Battlefield.CreaturScripts {
+    public class UniqueCreature : CreatureBase
     {
-        foreach (Ability ability in abilities)
+        [Header("Material Info")]
+        public Material[] mat;
+        [HideInInspector]
+        public Renderer currentMat;
+        float a = .05f;
+        float b;
+        int highlightColor;
+        [HideInInspector]
+        public UnitAbilitiesContainer abilityContainer;
+        [HideInInspector]
+        public UnitStats stats;
+        [Header("Virtual Camera Info")]
+        public CinemachineVirtualCamera currentCamera;
+        public Animator animator;
+
+        void Start()
         {
-            ability.Initialize(transform.gameObject);
+            health = maxHealth;
+
+            currentMat = GetComponent<Renderer>();
+            abilityContainer = GetComponent<UnitAbilitiesContainer>();
+            stats = GetComponent<UnitStats>();
         }
 
-        health = maxHealth;
 
-        currentMat = GetComponent<Renderer>();
-    }
-
-    public void UseAbility(int i, GameObject target)
-    {
-        anim.SetTrigger("Attack");
-        switch (abilities[i].TypeOfActionRequired)
+        /// <summary>
+        /// This method chnages the highlighting of a unit based on an integr.  3 = no highlight, 1 = blue, 2 = red.
+        /// When highlighting a unit dont forget to set their material back to normal using glow(3). 
+        /// This function might be removed in the future in favor of using a free store asset. 
+        /// </summary>
+        public void Glow(int glow)
```

Comment:
I'm most likely going to change this so that it uses a similar system to the asset that the other team mentioned to highlight objects instead of changing the material